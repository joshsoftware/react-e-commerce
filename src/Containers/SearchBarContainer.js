import React, { useState } from 'react';
import SearchBar from '../Components/SearchBar';
import product from '../Product';
const SearchBarContainer = () => {
  const [search, setSearch] = useState('');

  const filterdata = () => {
    const filterProduct = product.filter((product) => {
      return product.category.toLowerCase().includes(search.toLowerCase());
    });

    return filterProduct;
  };
  return <SearchBar placeholder={'search'} filterProduct={filterdata} setSearch={setSearch} />;
};
export default SearchBarContainer;
