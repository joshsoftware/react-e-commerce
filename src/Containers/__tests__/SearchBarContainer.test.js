import React from 'react';
import { render } from '@testing-library/react';
import SearchBarContainer from '../SearchBarContainer';

describe('SearchBarContainer ', () => {
  it('Must return entire SearchBarContainer', () => {
    const { asFragment } = render(<SearchBarContainer />);

    expect(asFragment()).toMatchSnapshot();
  });
});
