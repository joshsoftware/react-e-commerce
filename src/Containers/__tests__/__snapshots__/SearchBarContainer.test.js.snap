// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarContainer  Must return entire SearchBarContainer 1`] = `
<DocumentFragment>
  <input
    class="form-control"
    placeholder="search"
    type="text"
    value=""
  />
  <button
    class="btn btn-secondary"
    type="button"
  >
    Search
  </button>
</DocumentFragment>
`;
